### PR TITLE
Fs 2181 dashboard

### PR DESCRIPTION
**Pull request by srh-sloan**
Updates the dashboard to allow display of multiple rounds, which can be either open or closed. Displays as per prototype: https://fsd-pre-award.herokuapp.com/s31-application/window-open

### CODE REVIEW

File: app/default/account_routes.py [hunk 1]
```diff
@@ -1,67 +1,109 @@
...
+                past_submission_deadline = (
+                    current_datetime_after_given_iso_string(round["deadline"])
+                )
+                not_yet_open = current_datetime_before_given_iso_string(
+                    round["opens"]
+                )
+                apps_in_this_round = [
+                    app for app in applications if app.round_id == round_id
+                ]
+                if not not_yet_open:
```
Comment (tferns) [inline app/default/account_routes.py:52]: You could get rid of the `# noqa:E501`, double negative, and remove a block of indentation if you make use of the "return early pattern"

```suggestion
                if not_yet_open: continue
                
                for application in apps_in_this_round:
                    ...
```

File: app/default/account_routes.py [hunk 2]
```diff
@@ -1,67 +1,109 @@
...
 account_bp = Blueprint("account_routes", __name__, template_folder="templates")
 
+
+def build_application_data_for_display(applications: list[ApplicationSummary]):
+
+    application_data_for_display = {"funds": []}
+
+    count_of_applications_for_visible_rounds = 0
+    all_funds = get_all_funds()
+    if all_funds:
```
Comment (tferns) [inline app/default/account_routes.py:33]: Could get rid of a block of identation/make more readable if we make use of the "return early pattern" here

```suggestion
    application_data_for_display = {
        "funds": [],
        "total_applications_to_display": 0,
    }

    all_funds = get_all_funds()
    if not all_funds: return count_of_applications_for_visible_rounds 
    
    count_of_applications_for_visible_rounds = 0
    for fund in all_funds:
        ...
```

File: app/default/application_routes.py [hunk 3]
```diff
@@ -80,62 +128,69 @@ def tasklist(application_id):
...
+            form=form,
+            contact_us_email_address=round_data.contact_details[
+                "email_address"
+            ],
+            submission_deadline=round_data.deadline,
+            is_past_submission_deadline=current_datetime_after_given_iso_string(  # noqa:E501
+                round_data.deadline
+            ),
+        )
+    # return redirect(url_for("account_routes.dashboard"))
```
Comment (tferns) [inline app/default/application_routes.py:185]: Should this still be commented out?

File: app/templates/dashboard copy.html [hunk 4]
```diff
@@ -0,0 +1,118 @@
+{%- from "govuk_frontend_jinja/components/tag/macro.html" import govukTag -%}
```
Comment (tferns) [inline app/templates/dashboard copy.html:1]: Should this copy be committed?

File: app/templates/dashboard copy.html [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                'text': application.project_name if application.project_name else gettext('Untitled project')
+            },
+            {
+                'text':  govukTag({
+                    'text': application.status|snake_case_to_human ,
+                    'classes': status_class
+                })
+            },
+            {
+                'html': application.last_edited|datetime_format_short_month if application.last_edited
```
Comment (tferns) [inline app/templates/dashboard copy.html:60]: Not sure if the if is needed due to the changes you made to the filter:

https://github.com/communitiesuk/funding-service-design-frontend/blob/fead7cfe5a9f19559a8ba06017e387846450498b/app/filters.py#L11-L19

File: app/templates/partials/applications_table.html [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    {% for application in applications %}
+    {% set status_class %}
+    {% if is_past_submission_deadline %}
+        {% if application.status == 'SUBMITTED' %}
+            complete-tag
+        {% elif application.status == 'NOT_SUBMITTED' %}
+            complete-tag
+        {% endif %}
+    {% endif %}
+    {% if not is_past_submission_deadline %}
```
Comment (tferns) [inline app/templates/partials/applications_table.html:17]: Could probably just be an else:

```suggestion
    {% else %}
```

**Review by tferns — COMMENTED** [review]
Good work on this! Overall it all looks good.  I've left a few suggestions.  I'm hesitant to approve as there may be some accidentally committed files/accidentally commented out code

File: app/templates/partials/applications_table.html (hunk 6 above)
Comment (tferns) [reply app/templates/partials/applications_table.html:17]: Just realized this in the copied file...

File: app/templates/dashboard copy.html (hunk 5 above)
Comment (tferns) [reply app/templates/dashboard copy.html:60]: Just realized this in the copied file...

File: app/templates/dashboard.html [hunk 7]
```diff
@@ -4,115 +4,38 @@
...
-                    complete-tag
-                {%  elif application.status == 'NOT_STARTED' %}
-                    not-started-tag
-                {%  elif application.status == 'IN_PROGRESS' %}
-                    in-progress-tag-new
-                {%  elif application.status == 'READY_TO_SUBMIT' %}
-                    in-progress-tag-new
+                    {{ applications_table(round["applications"], round["is_past_submission_deadline"]) }}
+                {% if not round["is_past_submission_deadline"] %}
+                    {{ startApplicationButton(form, url_for('account_routes.new'), fund_id=fund["fund_id"], round_id=round["round_details"]["id"], existing_applications_count=round["applications"]|count) }}
```
Comment (tferns) [inline app/templates/dashboard.html:34]: @AbuNavsa This change might be relevant to what you asked about in slack

File: app/default/account_routes.py (hunk 1 above)
Comment (srh-sloan) [reply app/default/account_routes.py:52]: Thanks, good shout

File: app/default/application_routes.py (hunk 3 above)
Comment (srh-sloan) [reply app/default/application_routes.py:185]: Thanks - yes it should, it's left from before I changed the round open check to being a decorator.

File: app/templates/dashboard copy.html (hunk 4 above)
Comment (srh-sloan) [reply app/templates/dashboard copy.html:1]: Nope :)

**Review by tferns — APPROVED** [review]
Nice work! Looks good

**Pull request merged**